app/src/Bomberman.cpp

```diff
@@ -39,8 +68,110 @@ Bomberman::~Bomberman()
 void Bomberman::startGame()
 {
 	this->start();
-};
+}
+
+void Bomberman::handleMenuAction(MenuAction option)
+{
+	switch (option)
+	{
+	case MenuAction::StartGame:
+		this->menuState = MenuState::Playing;
+		if (!this->gameStarted)
+		{
+			this->gameStarted = true;
+			this->mainMenu.addOption(MenuItem(3, "Continue", false, MenuAction::StartGame));
+			this->mainMenu.resetSelected();
+		}
+		break;
+	case MenuAction::ToMainMenu:
+		this->menuState = MenuState::MainMenu;
+		break;
+	case MenuAction::ToSettingsMenu:
+		this->resolutionMenu.resetSelected();
+		this->controlsMenu.resetSelected();
+		this->menuState = MenuState::SettingsMenu;
+		break;
+	case MenuAction::ToResolutionMenu:
+		this->menuState = MenuState::ResolutionMenu;
+		break;
+	case MenuAction::ToControlsMenu:
+		this->menuState = MenuState::ControlsMenu;
+		break;
+	case MenuAction::SetResolution800:
+		if (this->resolution != Resolution::Default)
+		{
+			this->window->create(sf::VideoMode(800, 800), WINDOW_TITLE, sf::Style::Default, this->window->getSettings());
+			this->renderer.init();
+			this->resolution = Resolution::Default;
+		}
+		break;
+	case MenuAction::SetResolution1024:
+		if (this->resolution != Resolution::Medium)
+		{
+			this->window->create(sf::VideoMode(1024, 1000), WINDOW_TITLE, sf::Style::Default, this->window->getSettings());
+			this->renderer.init();
+			this->resolution = Resolution::Medium;
+		}
+		break;
+	case MenuAction::SetResolutionFullscreen:
+		if (this->resolution != Resolution::Fullscreen)
+		{
+			this->window->create(sf::VideoMode(1920, 1080), WINDOW_TITLE, sf::Style::Fullscreen, this->window->getSettings());
+			this->renderer.init();
+			this->resolution = Resolution::Fullscreen;
+		}
```

Comment:
A large portion of crashes can be prevented if the window is closed and rebuilt
'''case MenuAction::SetResolution800:
		if (this->resolution != Resolution::Default)
		{
			this->window->close();
			delete this->window;
			this->window = new sf::RenderWindow(sf::VideoMode(800, 800), WINDOW_TITLE, sf::Style::Default, settings);
			this->renderer.init();
			this->resolution = Resolution::Default;
		}
		break;
	case MenuAction::SetResolution1024:
		if (this->resolution != Resolution::Medium)
		{
			this->window->close();
			delete this->window;
			this->window = new sf::RenderWindow(sf::VideoMode(1024, 1024), WINDOW_TITLE, sf::Style::Default, settings);
			this->renderer.init();
			this->resolution = Resolution::Medium;
		}
		break;
	case MenuAction::SetResolutionFullscreen:
		if (this->resolution != Resolution::Fullscreen)
		{
			this->window->close();
			delete this->window;
			this->window = new sf::RenderWindow(sf::VideoMode(1920, 1080), WINDOW_TITLE, sf::Style::Fullscreen, settings);
			this->renderer.init();
			this->resolution = Resolution::Fullscreen;
		}'''